### PR TITLE
chore(StorageManager): typos and minor code cleanups

### PR DESCRIPTION
**Pull request by keturn**
Responding to code inspections and suggestions.

Some cleanup to reduce distractions while working on StorageManager for #4800.

Might have some conflicts with that branch, but both these branches are pretty small, shouldn't be too much trouble to resolve.

### CODE REVIEW

File: engine-tests/src/test/java/org/terasology/engine/persistence/internal/StorageManagerTest.java
Comment (jdrueckert) [inline engine-tests/src/test/java/org/terasology/engine/persistence/internal/StorageManagerTest.java:148]: is this still required or can it be removed entirely?

File: engine-tests/src/test/java/org/terasology/engine/persistence/internal/StorageManagerTest.java
Comment (jdrueckert) [inline engine-tests/src/test/java/org/terasology/engine/persistence/internal/StorageManagerTest.java:288]: same question

File: engine-tests/src/test/java/org/terasology/engine/persistence/internal/StorageManagerTest.java
Comment (jdrueckert) [inline engine-tests/src/test/java/org/terasology/engine/persistence/internal/StorageManagerTest.java:108]: why is this assertion no longer relevant?

Comment (pollend) [reply engine-tests/src/test/java/org/terasology/engine/persistence/internal/StorageManagerTest.java:108]: this should use the junit test function not assert.

File: engine-tests/src/test/java/org/terasology/engine/persistence/internal/StorageManagerTest.java
Comment (keturn) [reply engine-tests/src/test/java/org/terasology/engine/persistence/internal/StorageManagerTest.java:135]: This is the thing that spooks me about mocks.

Mockito told me that not all the mocks were used for all the tests. For instance, some of the tests don't use the NetworkManager mocks. A couple of these lines weren't used by _any_ of the tests—those are the ones I've commented out.

But I have a hunch that they were needed at some point or they wouldn't have been written. What if the next test needs them? Or what if something somewhere in the depths of one of these systems shifts which methods it calls when, and now this gets called again?

So I left it a little messy. With the intent that we'll come back to this after #5010 is merged and take another look at what kind of HeadlessEnvironment these tests need.

File: engine-tests/src/test/java/org/terasology/engine/persistence/internal/StorageManagerTest.java
Comment (keturn) [reply engine-tests/src/test/java/org/terasology/engine/persistence/internal/StorageManagerTest.java:108]: I think the intention of that assert was to make sure that we aren't re-using a directory that already has stuff in it. I took it out because it was already guaranteed unique by `@TempDir`, and because I removed this TempDir stuff because there's already TempDir stuff setting up the PathManager in TerasologyTestingEnvironment.

But your comment did prompt me to take another look, and I found that I had broken things here, I just hadn't gotten caught yet. (`global.dat` would be under `savePath`, not `tempHome`) This is a `BeforeEach`, but TTE was only doing the path setup `BeforeAll`.

I've fixed it now so every test gets its own save name, regardless of whether the PathManager has been reset between methods or not.

**Comment by keturn** [comment]
[Reactor: ReadWriteStorageManager engine (persistence)](https://trello.com/c/Be5t9Wtr/212-reactor-readwritestoragemanager-engine-persistence)

File: engine-tests/src/test/java/org/terasology/engine/persistence/internal/StorageManagerTest.java
Comment (jdrueckert) [reply engine-tests/src/test/java/org/terasology/engine/persistence/internal/StorageManagerTest.java:144]: in that case, please add a TODO there so that it's clear, why this is commented-out but not removed a.k.a what's unclear about it and what's necessary before it can be clarified and either re-introduced or removed.

File: engine-tests/src/test/java/org/terasology/engine/persistence/internal/StorageManagerTest.java
Comment (jdrueckert) [reply engine-tests/src/test/java/org/terasology/engine/persistence/internal/StorageManagerTest.java:284]: see above

**Review by jdrueckert — DISMISSED** [review]
I'll add my approval already now but please add TODO comments for the two commented-out lines before merging :)

**Comment by keturn** [comment]
Comments on comments, hmm?

Alright, how about this: I've un-commented those lines—the test's mockito strictness is already lowered to _lenient_, so that won't break anything—and I've added a TODO comment with an explanation for the lenient setting.

Thank you for the review! I'll merge once Jenkins approves this latest change.

**Pull request merged**
